### PR TITLE
CHI-2763 Resolve In-Progress Contacts: Close Tasks and Finalize Contacts

### PR DESCRIPTION
**Pull request by mythilytm**
## Description
- This PR introduces `Save and End` button, responsible for saving unresolved contacts left in progress. 
- A new permission `canEditInProgressContact` permission is added to control the Save and End button
- Refactor the In-Progress Banner with the 'Save and End' Button and display a dialog for saving the completing active calls and saving contacts
- Resolved Banner: Displays a blue banner when the contact is saved.

### Checklist
- [x] Corresponding issue has been [opened](https://tech-matters.atlassian.net/browse/CHI-2763)
- [ ] New tests added
- [ ] Feature flags added
- [x] Strings are localized
- [x] Tested for chat contacts
- [x] Tested for call contacts

### Other Related Issues

None

### Verification steps
- Ensure that this branch is deployed: https://github.com/techmatters/serverless/pull/698
- Start Flex(as a supervisor or owner) and start a call / chat in any of the channels
- Once the task is assigned, access the unfinalized contact via search with the In Progress banner
- You should see `Save and End.` Click on it and check that the behavior is aligned with the acceptance criteria.

### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

File: plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    };
+    await completeTaskAssignment(savedContact.taskId);
+    saveFinalizedContact(task, updatedContact);
+    setShowResolvedBanner(true);
+  };
+
+  const handleSaveAndEnd = async () => {
+    const status = await checkTaskAssignment(savedContact.taskId);
+
+    if (status === true) {
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx:76]:
```suggestion
    const isTaskAssigned = await checkTaskAssignment(savedContact.taskId);

    if (isTaskAssigned === true) {
```

Comment (GPaoloni) [reply plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx:76]: Also a question here: What happens if the task is in wrapup? :thinking:

File: plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      setIsDialogOpen(true);
+    } else {
+      await updateAndSaveContact();
+    }
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx:80]: How can anybody reach this? If dialog is open, you have to "click confirm", if it's not then this opens it.. Maybe this `else` statement is redundant? Or what am I missing?

File: plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx [hunk 3]
```diff
@@ -83,6 +83,8 @@ const BottomBar: React.FC<
   const handleSubmit = async () => {
     if (contactIsSaving || !hasTaskControl(task)) return;
 
+    console.log('>>> Submitting contact form', { task, contact, metadata, caseState });
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx:86]: :broom:

File: plugin-hrm-form/src/hrmConfig.ts [hunk 4]
```diff
@@ -85,6 +85,10 @@ const readConfig = () => {
     ]);
   const featureFlagsFromEnv = Object.fromEntries(featureFlagsFromEnvEntries);
   const featureFlagsFromServiceConfig: FeatureFlags = manager.serviceConfiguration.attributes.feature_flags || {};
+
+  // eslint-disable-next-line camelcase
+  featureFlagsFromServiceConfig.enable_generalized_search = false;
```
Comment (GPaoloni) [inline plugin-hrm-form/src/hrmConfig.ts:90]: :broom:

File: plugin-hrm-form/src/permissions/canEditInProgressContact.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ */
+import { getInitializedCan, PermissionActions } from '.';
+import { Contact } from '../types/types';
+import { getHrmConfig } from '../hrmConfig';
+
+const getCanEditInProgressContact = (contact: Contact, workerRoles: string[]): (() => boolean) => {
+  // const worker = Manager.getInstance().workerClient
+  console.log('>>> getCanEditInProgressContact', { workerRoles });
+  if (!contact.finalizedAt) {
+    // If the contact is a draft, we use the hardcoded rule that only its owner or creator can edit it
```
Comment (GPaoloni) [inline plugin-hrm-form/src/permissions/canEditInProgressContact.ts:24]: Why is this needed? :thinking: I mean why can't we use the same edit contact rules as for non-drafts?

**Review by GPaoloni — APPROVED** [review]
Code looks great to me! Thanks for organizing things better while in here :raised_hands: 

Leaving a couple of questions, shouldn't be anything blocking but is just to confirm I understand things.

File: plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx (hunk 2 above)
Comment (mythilytm) [reply plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx:80]: The dialog only opens when there is an active task (`assigned` task). In some cases, if the task is completed (triggered only `End` button in the agent-desktop by the owner) already, there is no need to show to dialog and we save directly.

File: plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx (hunk 1 above)
Comment (mythilytm) [reply plugin-hrm-form/src/components/contact/ContactInProgressBanners.tsx:76]: Good point. I realized I should include wrapup tasks in my first serverless function that checks the task status
